source/lac/petsc_vector_base.cc

```diff
@@ -294,9 +297,14 @@ namespace PETScWrappers
           }
         else
           {
-            std::vector<PetscInt> sorted_indices(
-              &array[end_index - ghost_start_index],
-              &array[n_elements_stored_locally]);
+            std::vector<PetscInt> sorted_indices;
+            sorted_indices.reserve(n_elements_stored_locally -
+                                   (end_index - ghost_start_index));
+            for (PetscInt i = end_index - ghost_start_index;
+                 i < n_elements_stored_locally;
+                 ++i)
+              sorted_indices.push_back(
+                static_cast<types::global_dof_index>(std::real(array[i])));
```

Comment:
Couldn't you do this with the `ConvertingIterator` as well? Just write it like this:
```
std::vector<PetscInt> sorted_indices(
              ConvertingIterator<...>(&array[end_index - ghost_start_index]),
              ConvertingIterator<...>(&array[n_elements_stored_locally]));
```